File: CPP Codes/16_SumOfDigitsOfString.cpp
#include <iostream>
#include <string>

using namespace std;

int sumOfDigits(string num) {
    int sum = 0;
    for (char c : num) {
        sum += c - '0';
    }
    return sum;
}

int getLucky(string s, int k) {
    string num;
    for (char c : s) {
        num += to_string(c - 'a' + 1);
    }

    while (k--) {
        num = to_string(sumOfDigits(num));
    }

    return stoi(num);
}

int main() {
    string s = "zbax";
    int k = 2;
    cout << getLucky(s, k) << endl;  // Output: 8
    return 0;
}

File: Python Codes/16_sumofdigitsofString.py
def sum_of_digits(num):
    return sum(int(digit) for digit in num)

def get_lucky(s, k):
    num = ''.join(str(ord(c) - ord('a') + 1) for c in s)
    
    for _ in range(k):
        num = str(sum_of_digits(num))
        
    return int(num)

# Example Usage
s = "zbax"
k = 2
print(get_lucky(s, k))  # Output: 8